py/gc.c

```diff
@@ -803,15 +803,36 @@ void gc_dump_info(void) {
            (uint)info.num_1block, (uint)info.num_2block, (uint)info.max_block, (uint)info.max_free);
 }
 
+#define DUMP_BLOCK_INFO (MICROPY_GC_DUMP_ALLOC_TABLE_MODE & MICROPY_GC_DUMP_ALLOC_TABLE_MODE_BLOCKS)
+#define DUMP_FRAGMENTATION_INFO (MICROPY_GC_DUMP_ALLOC_TABLE_MODE & MICROPY_GC_DUMP_ALLOC_TABLE_MODE_FRAG)
+#define NUM_FRAGMENTATION_CLASSES (20)
+
+#if DUMP_FRAGMENTATION_INFO
+STATIC void gc_update_fragmentation_stats(size_t n, size_t *frag_classes) {
+    for (size_t c = 1, i = 0; c < n; c <<= 1, ++i) {
+        frag_classes[i] += n / c;
```

Comment:
A comment that describes what this does would sure be nice. At least I can't just glance at the code and understand what it's doing... Maybe if the above `#define NUM_FRAGMENTATION_CLASSES` had a comment what frag classes are...

Reply:
Done